DFC.ServiceTaxonomy.GraphVisualiser/Queries/GetNodesCypherQuery.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System.Collections.Generic;
+using System.Text;
+using DFC.ServiceTaxonomy.Neo4j.Queries.Interfaces;
+using Neo4j.Driver;
+
+namespace DFC.ServiceTaxonomy.GraphVisualiser.Queries
+{
+    public class GetNodesCypherQuery : IQuery<string>
+    {
+        public string MatchPropertyName { get; }
+        public string MatchPropertyValue { get; }
+        public string SourcePropertyName { get; }
+        public string DestPropertyName { get; }
+        public Dictionary<long, INode> Nodes { get; }
+        public HashSet<IRelationship> Relationships { get; }
+        public long SelectedNodeId { get; set; }
+
+        public GetNodesCypherQuery(string matchPropertyName, string matchPropertyValue, string sourcePropertyName, string destPropertyName)
+        {
+            MatchPropertyName = matchPropertyName;
+            MatchPropertyValue = matchPropertyValue;
+            SourcePropertyName = sourcePropertyName;
+            DestPropertyName = destPropertyName;
+
+            Nodes = new Dictionary<long, INode>();
+            Relationships = new HashSet<IRelationship>();
+            SelectedNodeId = -1;
+        }
+
+        public void CheckIsValid()
+        {
+            // nothing to check, all properties are non-nullable
+        }
+
+        public Query Query
+        {
+            get
+            {
+                // $"match (n:ncs__JobProfile {{uri:\"{fetch}\"}})-[r]-(d) optional match (d)-[r1:esco__relatedEssentialSkill|:esco__relatedOptionalSkill|:ncs__hasUniversityLink|:ncs__hasUniversityRequirement]-(d1) return n, d, r, r1, d1";
+
+                var cypherQuery = new StringBuilder();
+
+                cypherQuery.Append($"MATCH (s {{{MatchPropertyName}:'{MatchPropertyValue}'}})");
+                cypherQuery.Append($"OPTIONAL MATCH (s {{{MatchPropertyName}:'{MatchPropertyValue}'}})-[r1]-(d1)");
```

Comment:
Can we remove the property match here, i.e. (`{{{MatchPropertyName}:'{MatchPropertyValue}'}}`)?

Reply:
Amended as suggested and it works